packages/dds/tree/src/simple-tree/core/types.ts

```diff
@@ -70,15 +86,23 @@ export interface TreeChangeEvents {
 	 * For remote edits, this event is not guaranteed to occur in the same order or quantity that it did in
 	 * the client that made the original edit.
 	 *
-	 * When it is emitted, the tree is guaranteed to be in-schema.
+	 * When the event is emitted, the tree is guaranteed to be in-schema.
 	 *
 	 * @privateRemarks
 	 * This event occurs whenever the apparent contents of the node instance change, regardless of what caused the change.
 	 * For example, it will fire when the local client reassigns a child, when part of a remote edit is applied to the
 	 * node, or when the node has to be updated due to resolution of a merge conflict
 	 * (for example a previously applied local change might be undone, then reapplied differently or not at all).
+	 *
+	 * TODO: defined and document event ordering (ex: bottom up, with nodeChanged before treeCHange on each level).
 	 */
-	nodeChanged(): void;
+	nodeChanged(
+		data: NodeChangedData &
```

Comment:
If I understand this correctly, for array nodes the API will still say there is an optional `<listenerArg>.changedProperties`, but at runtime it's always undefined? So there's no way to completely hide it from the listener's signature in that case?

Reply:
Removing it when passing in a node known to be an array node is possible, but that adds additional typing complexity to the signature and I don't think offers any value.

If someone knows their node is an array, seeing the changedProperties value and being able to read its docs doesn't seem like a bad thing. I don't imagine anyone will write code which would be able to be shorter or simpler if we hide that property from the array case. This is not true for making it never undefined for object and map nodes as such code would otherwise need to check for undefined before reading the set. Array specific code on the other hand probably will never need to check for it being a set before reading from the undefined (as there is nothing to read from an undefined value))